src/common/AmazonS3Backend.ts

```diff
@@ -304,11 +304,26 @@ export class AmazonS3Backend implements IBackend {
 
             // aws stores the part md5 in the ETag for multipart uploads
             if (partMD5 === data.Parts[i].ETag) {
-              // TODO add part to completed parts in upload
+
+              // modify current upload object if it exists
+              if (params.Body.s3upload) {
+
+                // ignore part if it has been or is being uploaded
+                if (params.Body.s3upload['completeInfo'][partNumber]) {
+                  return;
+                }
+
+                // update info for complete parts
+                params.Body.s3upload['completeInfo'][partNumber] = {ETag: data.Parts[i].ETag, PartNumber: partNumber};
+                params.Body.s3upload['totalUploadedBytes'] = params.Body.s3upload['totalUploadedBytes'] + data.Parts[i].Size;
+                params.Body.s3upload['doneParts'] = params.Body.s3upload['doneParts'] + 1;
```

Comment:
Seeing the following warnings:
```
[317, 38]: object access via string literals is disallowed
[318, 38]: object access via string literals is disallowed
[318, 83]: object access via string literals is disallowed
[319, 38]: object access via string literals is disallowed
[319, 74]: object access via string literals is disallowed
```

Recommend changing these lines to something like:
```
params.Body.s3upload.completeInfo[partNumber] = {
  ETag: data.Parts[i].ETag,
  PartNumber: partNumber
};
params.Body.s3upload.totalUploadedBytes += data.Parts[i].Size;
params.Body.s3upload.doneParts += 1;
```